TriblerGUI/widgets/graph_provider.py

```diff
@@ -10,25 +12,77 @@ class GraphProvider():
     Provides the matplotlib figure of the network.
     """
 
-    def provide_figure(self):
+    def __init__(self):
+        self.peers = None
+
+    def provide_figure(self, network_model):
         """
         Provide the matplotlib figure computed from the multichain data.
         TODO: add actual multichain data implementation, dummy data is inserted for now.
         
         :return: the matplotlib figure
         """
-        G = nx.Graph()
-        pos = [(2, 2), (1, 1), (2, 3), (3, 1)]
+        graph = nx.Graph()
+
+        display_information = network_model.retrieve_display_information()
+
+
+
+
+
+        """ Dummy data of a focus node and its peers """
+        up_down = [(50, 50), (91, 100), (52, 10), (20, 44), (1, 76), (17, 36)]
+        node_count = len(up_down)
+
+        # The focus node
+        center_x = 1
+        center_y = 1
+        focus_node = (center_x, center_y)
+
+        def polar_to_cartesian(base_x, base_y, radius, alpha):
+            return (base_x + radius * math.cos(math.pi / 2 - alpha),
+                    base_y - radius * math.sin(math.pi / 2 - alpha))
+
+        # Radius of first level neighbors
+        radius = 1
 
+        # Angle between neighbors
+        relative_angle = (2 * math.pi) / node_count
+
+        # Compute position of peers
+        self.peers = [polar_to_cartesian(center_x, center_y, radius, x * relative_angle) for x in range(node_count)]
+
+        def draw_node(pos, color):
+            """ Draw a networkx node with a specified color """
```

Comment:
Please start the contents of a multi-line comment on a new line and end sentences with a `.`.